### DISCUSSION TITLE
[Bug]Something to do with Plasma~1 projectile image historically is causing problems with some old branches

### DISCUSSION THREAD

**Discussion by Zitchas** [endless-sky/endless-sky#10761] [General]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When accessing old branches, especially from other people's forks, I get a warning about the Plasma~1 projectile image. When I look in my folder, I can't find such an image, since now they are plasma^1

![2024-11-06 What is up with the Plasma 1 projectile](https://github.com/user-attachments/assets/a9587cac-049e-43ff-a221-f4455198ccf0)

Both buttons lead to the same place: Back on whatever branch I happened to be on before switching.

This isn't the first time it has happened, and it's pretty much always when switching to an old branch. 

And just to be clear, there are no changes on my current branch, and this happens even with a complete clean ES git repository freshly downloaded from the site.

### Steps to Reproduce

1. Switch to a branch that is older than about a year.
2. See error

### Expected Behavior

Git should be able to handle all our changes and switch branches cleanly.

### Operating System

Windows 10 Pro

### Game Source

Other (please describe)

### Game Version

Git process, not game build

### Additional Information

The main thing I'd like to know is what happened differently with the plasma~1 rename? We've renamed, moved, and altered tons of files over the year, but something specific seems to be causing a problem with this one.

**Comment by Zitchas** [comment]
The image in question was renamed here: https://github.com/endless-sky/endless-sky/pull/10429

**Comment by warp-core** [comment]
This is an issue with Windows and GitHub Desktop.

Use git (not GitHub Desktop) to switch branches from the command line, possibly adding an argument to force it to carry out the switch. You may also need  You may also need to disable Windows 8.3 file name compatibility.

**Comment by Zitchas** [comment]
Even using git directly I ended up having to use --force to switch branches. If it is a problem with Windows & Win Desktop, why doesn't it trigger problems for any of the other dozens of images that have ~ in the name?

In any case, thanks for confirming that it is a problem.

**Reply by warp-core** [reply]
The name "plasma\~1.png" matches the format of a Windows 8.3 short file name. On systems where compatibility with this type of file name is enabled, names such as the recently added "plasma discharge+.png" are considered equivalent to "plasma~1.png".

**Reply by Zitchas** [reply]
Yes, I have confirmed that this is the case. (manually renamed the file to `plasma~1` again and it asked if I would like to rename to something else because there's already a `plasma~1` in the form of the discharge that you mention there. It was amusing to see.
